### PR TITLE
Refactor Account Deletion Process in Security Component

### PR DESCRIPTION
**Pull request by BruhLemma-Yadecha**
This pull request refactors the account deletion process in the `Security` component. The previous implementation utilized axios directly to handle user account deletion, while the new approach leverages a dedicated service function for cleaner and more maintainable code. Additionally, feedback mechanisms for the user have been improved via toast notifications. Below are the key changes made in this PR:

### CODE REVIEW

File: src/pages/ProfilePage/Security/Security.jsx [hunk 1]
```diff
@@ -22,24 +23,21 @@ const Security = () => {
...
+        localStorage.clear();
+        navigate('/');
+        window.location.reload();
+      })
+      .catch((error) => {
+        console.error('Error deleting account: ', error);
+        toast.error('Error deleting account');
+        setTimeout(() => {
+          navigate('/');
+        }, 3000);
```
Comment (amhajja) [inline src/pages/ProfilePage/Security/Security.jsx:39]: I'm curious, why did you add this?

File: src/pages/ProfilePage/Security/Security.jsx [hunk 2]
```diff
@@ -22,24 +23,21 @@ const Security = () => {
...
+  const triggerDeleteAccount = () => {
     setOpenDialog(false);
-    try {
-      const response = await axios.delete('https://api.edukona.com/delete-user', {
-        headers: { Authorization: `Token ${localStorage.getItem('token')}` },
+    deleteUser()
+      .then(() => {
+        localStorage.clear();
+        navigate('/');
+        window.location.reload();
```
Comment (amhajja) [inline src/pages/ProfilePage/Security/Security.jsx:32]: This might be unnecessary here!

**Review by amhajja — APPROVED** [review]
Left a few comments!

File: src/pages/ProfilePage/Security/Security.jsx (hunk 1 above)
Comment (BruhLemma-Yadecha) [reply src/pages/ProfilePage/Security/Security.jsx:39]: If we encounter an error while deleting a user, redirecting immediately would mean the toast showing an error had occured is never seen. This gives a small amount of time for that to pop up.

Comment (amhajja) [reply src/pages/ProfilePage/Security/Security.jsx:39]: With navigate, I don't think that's true; the toast will show up. I was also wondering if we need to redirect them to the root to begin with(?)

**Review by amhajja — APPROVED** [review]
Looks great!
